`DeserializeDict` fails if struct field is named 'key'

The following fails

```Rust
#[derive(DeserializeDict, Type)]
#[zvariant(signature = "a{sv}")]
pub struct OwnedProperties {
    key: String,
    val: OwnedValue,
}
```

like so: 
```Term
error[E0308]: mismatched types
 --> src/main.rs:7:10
  |
7 | #[derive(DeserializeDict, Type)]
  |          ^^^^^^^^^^^^^^^
  |          |
  |          expected `&str`, found `Option<_>`
  |          expected due to the type of this binding
  |
  = note: expected reference `&str`
                  found enum `std::option::Option<_>`
  = note: this error originates in the derive macro `DeserializeDict` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider using `Option::expect` to unwrap the `std::option::Option<_>` value, panicking if the value is an `Option::None`
  |
7 | #[derive(DeserializeDict.expect("REASON"), Type)]
  |  
```

Problem goes away as soon as the field 'key' is renamed to something else.

--- a/zvariant_derive/src/dict.rs
+++ b/zvariant_derive/src/dict.rs
@@ -196,8 +196,8 @@ pub fn expand_deserialize_derive(input: DeriveInput) -> Result<TokenStream, Erro
                         #( let mut #fields = ::std::default::Default::default(); )*
 
                         // does not check duplicated fields, since those shouldn't exist in stream
-                        while let ::std::option::Option::Some(key) = access.next_key::<&str>()? {
-                            match key {
+                        while let ::std::option::Option::Some(__key) = access.next_key::<&str>()? {
+                            match __key {
                                 #(#entries)*
                             }
                         }

--- a/zvariant_derive/tests/tests.rs
+++ b/zvariant_derive/tests/tests.rs
@@ -99,3 +99,16 @@ fn issues_311() {
         pub sinr: Option<i32>,
     }
 }
+
+#[test]
+#[ignore]
+fn issues_1252() {
+    // Issue 1252: Naming a field `key` in a dict struct causes a conflict with variables created by
+    // `DeserializeDict` macro, ending up with a strange error.
+    #[derive(DeserializeDict, Type)]
+    #[zvariant(signature = "a{sv}")]
+    pub struct OwnedProperties {
+        key: String,
+        val: OwnedValue,
+    }
+}

